CHANGELOG.md

```diff
@@ -1,3 +1,9 @@
+# unreleased
+
+## Bugfixes
+
+- Using faccess library to detect executable files, see #1106 (@ptipiak)
```

Comment:
This is interesting. I'm not entirely sure if I would classify this as a bug. The previous behavior was technically still in alignment with the documentation. The problem is that the definition of what it means for a file to be "executable" was not well defined. 

Speaking of which, it would probably still be good to better document what is meant by "executable" here. But now clarify that executable means "executable by the current effective user".

Reply:
I wasn't exatcly sure either how to classify it, since the goal is to implement every feature of the original find, I assume correcting deviation would be considered bugfix.

Reply:
I think "Using faccess library" is just an implementation detail that isn't necessary for the changelog.  It would be better to focus on how the change affects users, something "`--type executable` on Unix-like systems now checks whether a file is executable by the current user"

Reply:
> since the goal is to implement every feature of the original find, I assume correcting deviation would be considered bugfix.

Just to clarify: It is *not* `fd`s goal to implement every feature of find. But in terms of behavior, we certainly only want to deviate from `find` if there are really good reasons to do so.

For this particular feature (`--type executable`), it was really more of an oversight on my part. So both bugfix or change seem fine to me. Let's keep it under changes now.